src/Alert/BannerAlert.js

```diff
@@ -1,37 +0,0 @@
-import PropTypes from 'prop-types';
-
-import AlertContent from './AlertContent';
-import Box from '../Box';
-import IntentAlert from './IntentAlert';
-import INTENTS from './intents';
-
-const conditionalProps = (intent) => {
-  const props = intent
-    ? {
-        bg: `intent.${intent}.lighter`
-      }
-    : {};
-
-  return props;
-};
-
-const BannerAlert = ({ intent, ...props }) => (
-  <IntentAlert
-    Component={Box}
```

Comment:
I had no luck getting `IntentAlert`'s polymorphic behavior to typecheck properly, so I ended up inlining the passed components (`Box` here, `Card` and `Pane` elsewhere) and factoring out a `borderStyles` helper for the shared styling.